### DISCUSSION TITLE
I can't update a block of text after it's creation

### DISCUSSION THREAD

**Discussion by MarcoZucc** [codex-team/editor.js#1979] [Q&A]
I have a problem when I try to update a block of text after its creation, this is my code:

```
`$(document).ready(function(){

	var text_to_insert = [
        {
            "type" : "paragraph",
            "data" : {
                "text" : "Initial text abcd"
            }
        }];

var text_to_insert_afeter_ceckbox = [    
        {
            "type" : "paragraph",
            "data" : {
                "text" : "ABCD after ceckbo"
            }
        }];

var editor_IT = new EditorJS({

  holder : 'box_with_text',
  tools: {
     table: {
          class: Table,
          inlineToolbar: true,
          shortcut: 'CMD+ALT+T',
        },
  },
  data: {
  blocks: testo
       
  },
  onReady: function(){
   
      },
  onChange: function() {
        console.log('something changed');  
     //save function;
    
   },
    minHeight : 0
});


        $('input[type="checkbox"]').click(function(){
            if($(this).is(":checked")){
                console.log("Checkbox is checked.");
				editor_IT.blocks.insert(text_to_insert_afeter_ceckbox);
				
            }
            else if($(this).is(":not(:checked)")){
                console.log("Checkbox is unchecked.");
				editor_IT.blocks.insert(text_to_insert);
				
            }
        });

		
    });`
```

I tried to update the block of text using:

`editor_IT.blocks.insert(text_to_insert);`

But don’t wok
The aim would be is:

1) create a block of text with "acbd" for example

2) If user check or uncheck the checkbox I’m going to update the text in the editor area.

**Comment by williamhakim10** [comment]
It looks like you're passing an array of objects to `insert()` which isn't what the API specifies (see https://editorjs.io/blocks#insert). You could try `editor_IT.blocks.insert("paragraph", { text: "Initial text abcd" })`.

**Comment by MarcoZucc** [comment]
Ok the solution proposed by   @williamhakim10 works.

In my database i saved data in json format, for example:

```
{"time":1620914236050,"blocks":[
{"type":"paragraph","data":{"text":"example text or lorem ipsum"}}, {"type":"paragraph","data":{"text":"second line"}}
],"version":"2.20.1"} 
```

Can i use this string “as is” with insert method or I must extract data in blocks array and the insert in editor block by block ? 

There is a simple way to empty or destroy all text previously insert in the edit area and replace with new text ?
I have try with:

`editor.blocks.delete(0);`

and work fine but I need retrieve the total number of blocks and then delete them with a for loop.
Exist a method to reinitialize the object without content and insert the new data?

**Comment by williamhakim10** [comment]
@MarcoZucc you can initialize editorjs with that JSON blob, see https://editorjs.io/configuration#passing-saved-data. I don't see an easy way to delete all data and replace it with new data short of iterating over the blocks and deleting them and then inserting new data using the method in my previous comment.
